components/dashboard/src/admin/License.tsx

```diff
@@ -0,0 +1,215 @@
+                    <SolidCard>
+                        <div className="text-gray-600 dark:text-gray-200 py-4 flex-row flex font-semibold items-center">
+                            {statusMessage}
+                        </div>
+                        <p className="dark:text-gray-500 font-semibold">Registered Users</p>
+                        <span className="dark:text-gray-300 pt-1 text-lg">{license?.userCount || 0}</span>
+                        <span className="dark:text-gray-500 text-gray-400 pt-1 text-lg"> / {userLimit} </span>
+                        <p className="dark:text-gray-500 pt-2 font-semibold">License Type</p>
+                        <h4 className="dark:text-gray-300 pt-1 text-lg">{capitalizeInitials(license?.type || "")}</h4>
+                        <div className="flex justify-end bottom-2 relative">
+                            <button
+                                type="button"
+                                onClick={(e) => {
+                                    e.preventDefault();
+                                    window.location.href = "https://www.gitpod.io/self-hosted";
+                                }}
+                            >
+                                <div className="font-semibold">Compare Plans</div>
+                            </button>
+                        </div>
+                    </SolidCard>
+                </div>
+            </PageWithSubMenu>
+        </div>
+    );
+}
+
+function capitalizeInitials(str: string): string {
+    return str
+        .split("-")
+        .map((item) => {
+            return item.charAt(0).toUpperCase() + item.slice(1);
+        })
+        .join(" ");
+}
+
+function getSubscriptionLevel(license: LicenseInfo): ReactElement[] {
+    switch (license.plan) {
+        case "prod":
+        case "trial":
+            return professionalPlan(license.userCount || 0, license.seats, license.plan == "trial", license.validUntil);
+        case "community":
+            return communityPlan(license.userCount || 0, license.seats, license.fallbackAllowed);
+        default: {
+            return defaultMessage();
+        }
+    }
+}
+
+function licenseLevel(level: string): ReactElement {
+    return <p className="text-white dark:text-black font-bold pt-4"> {level}</p>;
+}
+
+function additionalLicenseInfo(data: string): ReactElement {
+    return <p className="dark:text-gray-500">{data}</p>;
+}
+
+function defaultMessage(): ReactElement[] {
+    const alertMessage = () => {
+        return (
+            <>
+                <div>Inactive or unknown license</div>
+                <div className="flex justify-right my-4 mx-1">
+                    <Alert fill="grey" className="h-8 w-8" />
+                </div>
+            </>
+        );
+    };
+
+    return [licenseLevel("Inactive"), additionalLicenseInfo("Free"), alertMessage()];
+}
+
+function professionalPlan(userCount: number, seats: number, trial: boolean, validUntil: string): ReactElement[] {
```

Comment:
In this code, the `validUntil` param is used for Trial licenses only. However, every license could have a limited validation (I expect that we usually issue 1-year licenses that need to be renewed). Right?